File: src/main/Java/boj_step21/boj25501.java
import java.io.*;

public class boj25501 {
        public static void main(String[] args) throws IOException {
            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
            BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
            int n = Integer.parseInt(reader.readLine());
            for (int i = 0; i < n; i++) {
                String s = reader.readLine();
                writer.append(isPalindrome(s).getX() + " " + isPalindrome(s).getY() + "\n");
            }
            writer.close();
        }
        private static Pair isPalindrome (String s) { return recursion(s, 0, s.length() - 1, 1); }
        private static Pair recursion(String s, int l, int r, int acc) {
            if (l >= r) return new Pair(1, acc);
            else if (s.charAt(l) != s.charAt(r)) return new Pair(0, acc);
            else return recursion(s, l+1, r-1, acc + 1);
        }
        private static class Pair {
            int x;
            int y;
            Pair (int x, int y) {
                this.x = x;
                this.y = y;
            }
            int getX() { return x; }
            int getY() { return y; }
        }
}

File: src/main/boj_step21/boj25501.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    repeat(r.readLine().toInt()) {
        val s = r.readLine()
        val result = isPalindrome(s)
        w.appendLine("${result.first} ${result.second}")
    }
    w.close()
}

private fun recursion(s: String, l: Int, r: Int, acc: Int): Pair<Int, Int> {
    if (l >= r) return Pair(1, acc)
    else if (s[l] != s[r]) return Pair(0, acc)
    else return recursion(s, l + 1, r - 1, acc + 1)
}

private fun isPalindrome(s: String): Pair<Int, Int> {
    return recursion(s, 0, s.length - 1, 1)
}